tools/recetox_aplcms/recetox_aplcms_hybrid.xml

```diff
@@ -59,16 +59,26 @@
 
     <expand macro="inputs">
         <expand macro="history_db"/>
-        <expand macro="noise_filtering"/>
+        <expand macro="noise_filtering">
+            <param name="min_exp" type="integer" min="1" value="2"
+            label="min_exp"
+            help="If a feature is to be included in the final feature table, it must be present in at least this number of samples." />
+        </expand>
         <expand macro="feature_detection"/>
         <expand macro="peak_alignment"/>
         <expand macro="weak_signal_recovery"/>
         <expand macro="output_format"/>
     </expand>
 
     <outputs>
-        <expand macro="unsupervised_outputs"/>
-        <data name="updated_known_table" format="parquet" label="${tool.name} updated_known_table on ${on_string}"/>
+        <expand macro="unsupervised_outputs">
+            <collection  name="extracted_features" type="list" label="${tool.name} extracted_features on ${on_string}">
+                <discover_datasets pattern="__designation__" directory="extracted" format="parquet" />
+            </collection >
+            <collection  name="corrected_features" type="list" label="${tool.name} corrected_features on ${on_string}">
+                <discover_datasets pattern="__designation__" directory="corrected" format="parquet" />
+            </collection >
+        </expand>

```

Comment:
Is this a way of combining a macro and other parameters in one block?

Reply:
Yes, you can expand your macros like this by placing `<yield />` where you need some parametrization. Here's another example from Planemo [docs](https://planemo.readthedocs.io/en/latest/writing_advanced.html#parameterizing-xml-macros-with-yield).